src/domain/core/src/utils/metadata_chain_comparator.rs

```diff
@@ -63,6 +63,7 @@ impl MetadataChainComparator {
 
         let Some(rhs_head) = rhs_head else {
             // LHS chain is unconditionally ahead - simply return all of its blocks
+            #[cfg_attr(target_pointer_width = "64", allow(clippy::cast_possible_truncation))]
```

Comment:
I think the root cause here is this structure, where `usize` could be changed to `u64`, and all uses, including listeners, need to adjust:

```
#[derive(Debug, Clone, Default)]
pub struct SyncPartyStats {
    /// Number of metadata blocks read from the party
    pub metadata_blocks_read: usize,
    /// Number of metadata blocks written to the party
    pub metadata_blocks_written: usize,
    /// Number of checkpoint files read from the party
    pub checkpoints_read: usize,
    /// Number of checkpoint files written to the party
    pub checkpoints_written: usize,
    /// Number of data files read from the party
    pub data_slices_read: usize,
    /// Number of data files written to the party
    pub data_slices_written: usize,
    /// Number of bytes read from the party
    pub bytes_read: usize,
    /// Number of bytes written to the party
    pub bytes_written: usize,
}
```

Reply:
Done